fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java

```diff
@@ -3432,6 +3434,11 @@ public JSONObject toJson() throws IOException {
                 if (attr == null) {
                     continue;
                 }
+                String defaultValue = VariableMgr.getDefaultValue(attr.name());
+                String currentValue = VariableMgr.getValue(this, new VariableExpr(attr.name(), SetType.SESSION));
+                if (defaultValue != null && defaultValue.equals(currentValue)) {
```

Comment:
If a user explicitly sets a value and it equals the default value, will it not be persisted? Then after an upgrade, if the default value changes, will the user's set value be 'changed' to the new one?